src/Engine/Graphics/Outdoor.cpp

```diff
@@ -1038,6 +1040,16 @@ bool OutdoorLocation::Load(const std::string &filename, int days_played,
                 }
             }
         }
+
+        // TODO(pskelton) : more accurate calculation?
+        // pskelton - many models have bounding z in wrong position testing this
+        if (model.vBoundingCenter.z < model.pBoundingBox.z1) {
```

Comment:
To be honest this makes no sense to me.

Shouldn't it be smth like this:

Vec3f topLeft = Vec3f(pBoundingBox.x1, pBoundingBox.y1, pBoundingBox.z1);
Vec3f bottomRight = Vec3f(pBoundingBox.x2, pBoundingBox.y2, pBoundingBox.z2);
model.vBoundingCenter = (topLeft + bottomRight) / 2.0f;
model.sBoundingRadius = (topLeft - model.vBoundingCenter).Length();

?

Reply:
That would be the more accurate calculation, will amend.
The original game treated this more like a cyclinder than a sphere I think. Hence why it was only checking left and right view in isbmodelvisible.